### DISCUSSION TITLE
New Feature: Sharing URL/ Link Management

### DISCUSSION THREAD

**Discussion by whitephoenix117** [photoprism/photoprism#3090] [Ideas]
Expose a master list of all sharing links similar to the "links" table in the DB. Allow for link administration

- Revocation of the URL
- Editing URL duration
- Changing the URL password
- Adding an admin comment to the link (could be name of who it was shared too)
- Showing use statistics for the URL (# of times used, first/ last use date, IP of connection(s))
- "Revoke all" button

**Comment by graciousgrey** [comment]
As far as I can see most points are part of other feature requests:

> xpose a master list of all sharing links similar to the "links" table in the DB

- https://github.com/photoprism/photoprism/issues/1330

>Revocation of the URL
>Editing URL duration

[This is already possible ](https://docs.photoprism.app/user-guide/share/#delete-sharing-link) and will also possible from the sharing link overview page (#1330)

>Changing the URL password

- https://github.com/photoprism/photoprism/issues/1428

>Showing use statistics for the URL (# of times used, first/ last use date, IP of connection(s))

- https://github.com/photoprism/photoprism/issues/2607

**Comment by graciousgrey** [comment]
I will convert this into a discussion as it is not a clearly defined feature request and contains some duplicates.
Feel free to leave comments and :+1: in the issues linked above :)
